docs/devel/api_changes.md

```diff
@@ -0,0 +1,237 @@
+# So you want to change the API?
+
+The Kubernetes API has two major components - the internal API structures and
+the versioned API structures.  The versioned APIs are intended to be stable,
+while the internal API is implemented to best reflect the needs of the
+Kubernetes code itself.  Clients should consume the versioned APIs exclusively.
+
+What this means for API changes is that you have to be somewhat thoughtful in
+how you approach changes, and that you have to touch a number of pieces to make
+a complete change.  This document aims to guide you through the process - not
+all API changes will need all of these steps.
+
+## On compatibility
```

Comment:
@bgrant0607 @smarterclayton @alex-mohr @brendanburns @erictune 

A random smattering of people to validate my definition of "compatible"